src/fstab-generator/fstab-generator.c

```diff
@@ -997,24 +1009,38 @@ static int parse_proc_cmdline_item(const char *key, const char *value, void *dat
         return 0;
 }
 
-static int determine_root(void) {
-        /* If we have a root hash but no root device then Verity is used, and we use the "root" DM device as root. */
+static int determine_device(char **what_p, char **hash_p, const char *name) {
```

Comment:
Please drop `_p` suffix. Also, `hash` is not modified in this function, so simply `const char *hash` should be ok.